### ISSUE TITLE
EDGE-16 web-Component suuuuper slow

### ISSUE THREAD

**Issue by AndyOGo** [axa-ch/patterns-library#369]
## Are you reporting a bug or a feature?

 - [x] Bug

## Expected Behavior

Web-Components should load blazing fast in every browser.

## Actual Behavior

**EDGE 16** has almost 10 seconds to render web components.

**Related issues:**
- Firefox slow #225 
- FOUC of web components #365 

## Steps to Reproduce the Problem

1. Open https://axa-ch.github.io/patterns-library/organisms.html in **EDGE 16**
2. https://www.browserstack.com/start#os=Windows&os_version=10&browser=Edge&browser_version=16.0&zoom_to_fit=true&full_screen=true&resolution=responsive-mode&url=https%3A%2F%2Faxa-ch.github.io%2Fpatterns-library%2Forganisms.html&speed=1&host_ports=google.com%2C80%2C0

![edge-16-super-slow](https://user-images.githubusercontent.com/914443/37970514-f6eba22c-31d3-11e8-9e94-97842047e53d.gif)

## Specifications

**TL/DR:** just paste results from https://www.whatsmybrowser.org/

Please provide any relevant information about your setup - os, browser, node, npm, etc.

 - Browser: EDGE
 - Browser Version: 16

**Note:** May all polyfilled browsers are affected.

**Comment by AndyOGo** [comment]
@LucaMele 
I found the reason why the polyfill is that sloooow. The algorithm is slow by nature 😱 
https://github.com/webcomponents/custom-elements#customelementspolyfillwrapflushcallback

> The polyfill gets slower as the size of your page and number of custom element definitons increases. You can use polyfillWrapFlushCallback to prevent redundant work.

> To avoid a potential memory leak, the polyfill does not maintain a list of upgrade candidates. This means that calling customElements.define causes a synchronous, full-document walk to search for elements with localNames matching the new definition. Given that this operation is potentially expensive and, if your page loads many custom element definitions before using any of them, highly redundant, an extra method is added to the CustomElementRegistry prototype - polyfillWrapFlushCallback.

They also suggest to use a custom `polyfillWrapFlushCallback` to mitigate this issue.

**Comment by AndyOGo** [comment]
@LucaMele 
Seems to become even a problem in chrome. see #460

**Comment by LucaMele** [comment]
closing as new plib version is live. If issue is with v2 still relevant, please re-open
